src/azure-cli/azure/cli/command_modules/backup/custom_base.py

```diff
@@ -87,13 +124,35 @@ def list_recovery_points(cmd, client, resource_group_name, vault_name, container
     if item.properties.backup_management_type.lower() == "azurestorage":
         return custom_afs.list_recovery_points(cmd, client, resource_group_name, vault_name, item, start_date,
                                                end_date, use_secondary_region)
+
     if item.properties.backup_management_type.lower() == "azureworkload":
         return custom_wl.list_wl_recovery_points(cmd, client, resource_group_name, vault_name, item,
                                                  start_date, end_date, use_secondary_region=use_secondary_region)
-
     return None
 
 
+def move_recovery_points(cmd, resource_group_name, vault_name, container_name, item_name, rp_name, source_tier,
+                         destination_tier, backup_management_type=None, workload_type=None):
+
+    items_client = backup_protected_items_cf(cmd.cli_ctx)
+    item = show_item(cmd, items_client, resource_group_name, vault_name, container_name, item_name,
+                     backup_management_type, workload_type)
+    custom_help.validate_item(item)
+
+    if isinstance(item, list):
+        raise ValidationError("Multiple items found. Please give native names instead.")
+
+    if item.properties.backup_management_type.lower() == "azureiaasvm":
+        return custom.move_recovery_points(cmd, resource_group_name, vault_name, item, rp_name, source_tier,
+                                           destination_tier)
+
+    if item.properties.backup_management_type.lower() == "azureworkload":
+        return custom_wl.move_wl_recovery_points(cmd, resource_group_name, vault_name, item, rp_name,
+                                                 source_tier, destination_tier)
+
+    raise InvalidArgumentValueError('This command is not supported for --backup-management-type AzureStorage.')
```

Comment:
`InvalidArgumentValueError` is for the case that the input value used for parameter is illegal. However, the problem here is that the parameters are not supported, so the error type `ArgumentUsageError` is more appropriate here, right?

Reply:
done